### DISCUSSION TITLE
Select all elements in data table

### DISCUSSION THREAD

**Discussion by r-delprete** [tusen-ai/naive-ui#4951] [Q&A]
Hi everyone, I want to select all elements in a data table, without using 'option' prop. Can someone help me?
The data table mentioned, is with pagination, so i would select all elements in all pages without 'option' prop, but with click on checkbox on header column

**Comment by jahnli** [comment]
`default-checked-row-keys`  ?

**Reply by r-delprete** [reply]
doesn't work.
I think I should write the option prop 'all' logic for checkbox selection, but I don't now how

**Comment by jahnli** [comment]
https://codesandbox.io/s/wild-violet-cstm7m?file=/src/Demo.vue

**Reply by r-delprete** [reply]
sorry but I don't want default checked rows.
My scenario is:
I have a data-table with more pages. In this data-table, an user can select rows by checkboxes. When the user decide to select ALL the rows in the table, the header checkbox select only the rows in the current page. But I want to select all in all pages when the header checkbox is clicked

**Comment by jahnli** [comment]
I can do that with default-checked-row-keys or checked-row-key, It depends on the scenario you're using

ref

https://www.naiveui.com/zh-CN/os-theme/components/data-table#select.vue

**Comment by jahnli** [comment]
https://codesandbox.io/s/pensive-colden-3fwq7h?file=/src/Demo.vue

I'm not sure if this is the effect you need, and if not, maybe I don't understand your needs well enough

@roccodelprete

**Reply by r-delprete** [reply]
thank you anyway :)
